### DISCUSSION TITLE
Assert test class for top-level function (composable)

### DISCUSSION THREAD

**Discussion by chrisbanes** [LemonAppDev/konsist#817] [Ask a Question]
### Wanted end result.

This is a follow on from #627. What if we want to assert that a top-level function was a relevant test?  The use case I have is actually the same one described in #627. If I have a `@Composable fun MyFeature(...)`, how can I assert that there is a test class called `MyFeatureScreenshotTest`?

AFAICT `hasTestClass` is only available for classes, not functions.

**Comment by nataliapeterwas** [comment]
We don't have a dedicated method to check if a function has a test.

I think this solution should help you:
```kotlin
 val testClasses = Konsist
    .scopeFromTest()
    .classes()
    .map { it.name }

Konsist
    .scopeFromProject()
    .functions()
    .withAnnotationNamed("Composable")
    .assertTrue { function ->
        testClasses.any {  testClassName ->
            testClassName == function.name + "ScreenshotTest"
        }
    }
```
